openlcs/libs/sc_handler.py

```diff
@@ -0,0 +1,130 @@
+        Get the srpm components list from source container itself. The
+        component value is almost same as the srpm component from corgi.
+        The value of srpm component from corgi is as following:
+        {
+            'uuid': 'uuid',
+            'type': 'SRPM',
+            'name': 'libcom_err',
+            'version': '1.45.6',
+            'release': '2.el8',
+            'license': '',
+            'arch': '',
+            'synced': False
+        }
+        @params: srpm_dir, destination directory for srpm files' dir
+        """
+        components = []
+        if os.path.isdir(srpm_dir):
+            srpms = glob.glob(srpm_dir + "/*.src.rpm")
+            for srpm in srpms:
+                srpm = srpm.split('/')[-1]
+                rpm_nvr = srpm.replace('.src.rpm', '')
+                nvr = koji.parse_NVR(rpm_nvr)
+                component = self.get_component_flat(nvr, 'SRPM')
+                components.append(component)
+        else:
+            err_msg = f'Not found {srpm_dir}'
+            raise RuntimeError(err_msg) from None
+        return components
+
+    def get_source_of_srpm_component(self, srpm_dir, nvr):
+        """
+        Get the source file or srpm
+        """
+        srpm_file = nvr + '.src.rpm'
+        srpm_filepath = os.path.join(srpm_dir, srpm_file)
+        if os.path.isfile(srpm_filepath):
+            return srpm_filepath
+        else:
+            err_msg = f'Not found {srpm_filepath}'
+            raise RuntimeError(err_msg) from None
+
+    def get_misc_component(self, misc_dir, sc_nvr):
+        """
+        Get the metadata component of the source container.
+        After unpacking the source container, it has some matedata files.
+        The metadata files need to be gathered as one misc component.
+        @params: misc_dir, destination directory for source container metadata
+        @params: sc_nvr, source container nvr
+        """
+        components = []
+        if os.path.isdir(misc_dir):
+            nvr = koji.parse_NVR(sc_nvr)
+            nvr['name'] = nvr.get('name') + '-metadata'
+            component = self.get_component_flat(nvr, 'UNKNOWN')
+            components.append(component)
+        else:
+            err_msg = f'Not found {misc_dir}'
+            raise RuntimeError(err_msg) from None
+        return components
+
+    def get_source_of_misc_component(self, misc_dir, nvr):
+        """
+        Compress the metadata files to tar file
+        """
+        # Named the source tar file that unified with misc component
+        tar_file = nvr + '.tar.gz'
+        # Compress the metadata files to misc nvr tar file
+        try:
+            tar_filepath = os.path.join(os.path.dirname(misc_dir), tar_file)
+            compress_source_to_tarball(tar_filepath, misc_dir)
+        except RuntimeError as err:
+            raise RuntimeError(err) from None
+        return tar_filepath
+
+    def get_container_components(self, srpm_dir, misc_dir, sc_nvr):
```

Comment:
this `SourceContainerHandler.get_container_componnets` accepts `srpm_dir` and `misc_dir` so you're assuming it contains "srpms" and all "srpms" are inside "srpm_dir".

I'm worrying about whether it's generic enough.

Reply:
If there are no srpms in the srpm_dir, the type 'SRPM' of the component is None. So no task will be forked. Do you mean to give a default value to srpm_dir?

Reply:
I mean this param indicates that to get source container components, you'll need `srpm_dir` - you're assuming source container always consists of srpms. I probably am wrong, what I understand is that this `srpm_dir` makes sense only for containers that has srpm components.

Reply:
In the container [unpack function](https://github.com/RedHatProductSecurity/openlcs/blob/main/openlcs/libs/unpack.py#L227), it always creates three directories, misc_dir, rpm_dir, rc_dir.  This function, get_container_components(self, srpm_dir, misc_dir, sc_nvr), I guess it should be updated to get_container_components(self, srpm_dir, **rc_dir,** misc_dir, sc_nvr) after adding the remote source part.  If there are no srpm components, the SRPM components will be None. If there are no remote source components, the remote source components will be None. 
If in the sc_handler, the srpm_dir is for SRPM, and the rc_dir is for different remote source type components, the generic issue won't be important.

Additional info:
The components structure: 
components = {
    'SRPM': srpm_components,
    'GOLANG/NPM/...': rc_components,
    'CONTAINER_IMAGE': container_component
}